bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,234 @@
+                            YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.CONNECTION_FAILED;
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                    "Connection lost no ping from YIO DOCK");
+                            updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+                            pollingJob.cancel(true);
+                        }
+                    } catch (InterruptedException e) {
+                        logger.warn("Error during initializing the WebSocket polling Thread {}", e.toString());
+                    }
+                }
+            };
+            try {
+                pollingJob = scheduler.scheduleWithFixedDelay(heartbeatpolling, 0, 30, TimeUnit.SECONDS);
+            } catch (Exception e) {
+                updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+                logger.warn("Error during starting the WebSocket polling Thread {}", e.toString());
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public void dispose() {
+        updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+        pollingJob.cancel(true);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (YIODOCKRECEIVERSWITCH.equals(channelUID.getIdWithoutGroup())) {
+            if (YIOREMOTEHANDLESTATUS_actualstatus.equals(YIOREMOTEHANDLESTATUS.AUTHENTICATED)) {
+                if (command instanceof RefreshType) {
+                    // TODO: handle data refresh
+
+                }
+
+                if (command.toString().equals("ON")) {
+                    logger.debug("YIODOCKRECEIVERSWITCH ON procedure: Switching IR Receiver on");
+                    YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.IRRECEIVERON, "");
+                } else if (command.toString().equals("OFF")) {
+                    logger.debug("YIODOCKRECEIVERSWITCH OFF procedure: Switching IR Receiver off");
+                    YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.IRRECEIVEROFF, "");
+                } else {
+                    logger.debug("YIODOCKRECEIVERSWITCH no procedure");
+                }
+
+            } else {
+                logger.warn("YIOremoteDockHandler not authenticated");
+            }
+
+        } else if (YIODOCKSENDIRCODE.equals(channelUID.getIdWithoutGroup())) {
+            if (YIOREMOTEHANDLESTATUS_actualstatus.equals(YIOREMOTEHANDLESTATUS.AUTHENTICATED)) {
+                if (command instanceof RefreshType) {
+                    // TODO: handle data refresh
+                    // logger.warn("YIOremoteDockHandler not authenticated");
+                }
```

Comment:
If I see correctly this is a trigger channel. So a refresh doesn't need to be handled.

Reply:
removed